File: Codeforces/round_797/task_B.py
# https://codeforces.com/contest/1690/problem/B

from math import inf

INF = (-1) * inf

def solving(a, b, n):
    c = []
    for i in range(n):
        if b[i] != 0:
            ans = a[i] - b[i]
        else:
            ans = INF
        c.append(ans)
    new_set = set(c)
    if len(new_set) == 1 and c[0] >= 0:
        ans = "YES"
    elif len(new_set) == 1 and INF in new_set:
        ans = "YES"
    elif len(new_set) == 2 and  INF in new_set:
        ans = "YES"
    else:
        ans = "NO"
    return ans

def main():
    t = int(input())
    answer = []
    for _ in range(t):
        n = int(input())
        a = list(map(int, input().split()))
        b = list(map(int, input().split()))
        answer.append(solving(a, b, n))
    answer = "\n".join(answer)
    print(answer)

if __name__ == "__main__":
    main()

File: coding/Codeforces/ed_round_144/task_B.cpp
// https://codeforces.com/contest/1796/problem/B
// Created by ruslan515 on 28.02.23.
//

#include <bits/stdc++.h>

using namespace std;

string solving() {
    string ans = "NO\n";
    string a, b;
    cin >> a >> b;
    int n = a.size();
    int m = b.size();
    if (a[0] == b[0]) {
        ans = "YES\n";
        ans += a[0];
        ans += "*\n";

    } else if (a[n - 1] == b[m - 1]) {
        ans = "YES\n";
        ans += "*";
        ans += a[n - 1];
        ans += "\n";

    } else {
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                if (a[i] == b[j] && (i < n - 1 && j < m - 1) && (a[i + 1] == b[j + 1])) {
                    ans = "YES\n";
                    ans += "*";
                    ans += a[i];
                    ans += a[i + 1];
                    ans += "*\n";
                    break;
                }
            }
        }
    }


    return ans;
}


int main() {
    ios::sync_with_stdio(false);
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    int t;
    cin >> t;
    for (int i = 0; i < t; ++i) {
        cout << solving();
    }

    return 0;
}